CMAKE_INSTALL_PREFIX being ignored

<!-- Before reporting: search existing issues and check the FAQ. -->

- `nvim --version`: NVIM v0.4.0-413-g9ef8d0d
- Vim (version: ) behaves differently? Yes, can not recall version which worked earlier.
- Operating system/version: Linux Mint 18.3
- Terminal name/version: xterm (322)
- `$TERM`: xterm-256color

### Steps to reproduce 
run `make CMAKE_INSTALL_PREFIX=$HOME/Work/Tools/neovim/ MAKE_BUILD_TYPE=Release`
then `make install` 

Installation will fail with customized prefix. cmake ignores CMAKE_INSTALL_PREFIX.

### Actual behaviour

```
 make install

error: CMAKE_INSTALL_PREFIX '/home/dan/Work/Tools/neovim/' does not match cached value '/usr/local'
       Run this command, then try again:
         cmake build -DCMAKE_INSTALL_PREFIX=/home/dan/Work/Tools/neovim/
Makefile:169: recipe for target 'checkprefix' failed
make: *** [checkprefix] Error 1
```

Tried:
```
make CMAKE_INSTALL_PREFIX=/home/dan/Work/Tools/neovim/ CMAKE_BUILD_TYPE=Release CMAKE_INSTALL_MANDIR=/home/dan/Work/Tools/neovim/
```
This failed with:
```
 make install

error: CMAKE_INSTALL_PREFIX '/home/dan/Work/Tools/neovim/' does not match cached value '/usr/local'
       Run this command, then try again:
         cmake build -DCMAKE_INSTALL_PREFIX=/home/dan/Work/Tools/neovim/
Makefile:169: recipe for target 'checkprefix' failed
make: *** [checkprefix] Error 1
```
Also w/o `CMAKE_INSTALL_MANDIR`, this failed with very first error above.

### Expected behaviour

`make install` will install binaries, man pages and scripting into CMAKE_INSTALL_PREFIX directory.

> cmake caches variables in `build/CMakeCache.txt`. You probably ran `make` without `CMAKE_INSTALL_PREFIX` first and thus cmake cached the default location of `/usr/local`.
> 
> Another note: `CMAKE_INSTALL_PREFIX` is a cmake thing and only works with `make` because our `Makefile` explicitly passes it to cmake. That won't work for `CMAKE_INSTALL_MANDIR`, though.
> 
> So, the correct way to do what you had in mind:
> 
> ```
> $ rm -r build
> $ make CMAKE_INSTALL_PREFIX=/home/dan/Work/Tools/neovim CMAKE_EXTRA_FLAGS="-DCMAKE_INSTALL_MANDIR=/home/dan/Work/Tools/neovim"
> ```
> 
> Alternatively, if you intend to build more often from source, you can put these things in a `local.mk` in the root directory:
> 
> ```
> CMAKE_INSTALL_PREFIX = ${HOME}/Work/Tools/neovim
> CMAKE_EXTRA_FLAGS = -DCMAKE_INSTALL_MANDIR=${HOME}/Work/Tools/neovim
> ```
> 
> Afterwards, only `make && make install` is needed.
> 
> If this works for you, we can probably improve that error message.

Thank you for elaborated reply. Unfortunately none of the steps you described lead to the successful `make install`. After modifying variables as per your recommendation I ended up with following:
```
make install
ninja  -C .deps
ninja: Entering directory `.deps'
[2/2] cd /home/dan/Work/Sources/neovim/.deps && /usr/bin/cmake -E touch .third-party
ninja  -C build
ninja: Entering directory `build'
ninja: no work to do.
ninja  -C build install
ninja: Entering directory `build'
[0/1] Install the project...
-- Install configuration: "Debug"
-- Installing: /home/dan/Work/Tools/neovim/man1/nvim.1
-- Installing: /usr/local/bin/nvim
CMake Error at src/nvim/cmake_install.cmake:78 (file):
  file INSTALL cannot copy file
  "/home/dan/Work/Sources/neovim/build/bin/nvim" to "/usr/local/bin/nvim".
Call Stack (most recent call first):
  cmake_install.cmake:85 (include)
  


FAILED: CMakeFiles/install.util 
cd /home/dan/Work/Sources/neovim/build && /usr/bin/cmake -P cmake_install.cmake
ninja: build stopped: subcommand failed.
Makefile:143: recipe for target 'install' failed
make: *** [install] Error 1
```
So judging based on very quick closure from @justinmk it looks like there are more pressing issues in this project so I am going back to my cave and learn a bit about `cmake` and will try to bring a fix for this as I am not the only one who is experiencing this.

Thank you again for your help @mhinz !

--- a/Makefile
+++ b/Makefile
@@ -7,7 +7,6 @@ filter-true = $(strip $(filter-out 1 on ON true TRUE,$1))
 
 CMAKE_PRG ?= $(shell (command -v cmake3 || echo cmake))
 CMAKE_BUILD_TYPE ?= Debug
-CMAKE_FLAGS := -DCMAKE_BUILD_TYPE=$(CMAKE_BUILD_TYPE)
 # Extra CMake flags which extend the default set
 CMAKE_EXTRA_FLAGS ?=
 
@@ -16,6 +15,7 @@ CMAKE_EXTRA_FLAGS ?=
 #   - `checkprefix` target checks that it matches the CMake-cached value. #9615
 CMAKE_INSTALL_PREFIX ?= $(shell echo $(CMAKE_EXTRA_FLAGS) | 2>/dev/null \
     grep -o 'CMAKE_INSTALL_PREFIX=[^ ]\+' | cut -d '=' -f2)
+CMAKE_FLAGS := -DCMAKE_BUILD_TYPE=$(CMAKE_BUILD_TYPE) -DCMAKE_INSTALL_PREFIX=$(CMAKE_INSTALL_PREFIX)
 
 BUILD_TYPE ?= $(shell (type ninja > /dev/null 2>&1 && echo "Ninja") || \
     echo "Unix Makefiles")
@@ -166,7 +166,7 @@ appimage-%:
 lint: check-single-includes clint testlint lualint
 
 checkprefix:
-	@cached_prefix=$$($(CMAKE_PRG) -L -N build | 2>/dev/null grep 'CMAKE_INSTALL_PREFIX' | cut -d '=' -f2); \
+	@cached_prefix=$$("$(CMAKE_PRG)" -L -N build | 2>/dev/null grep 'CMAKE_INSTALL_PREFIX' | cut -d '=' -f2); \
 	if [ -n "$(CMAKE_INSTALL_PREFIX)" ] && [ -n "$$cached_prefix" ] && ! [ "$(CMAKE_INSTALL_PREFIX)" = "$$cached_prefix" ]; then \
 		printf "\nerror: CMAKE_INSTALL_PREFIX '$(CMAKE_INSTALL_PREFIX)' does not match cached value '%s'\n" "$$cached_prefix"; \
 		printf "       Run this command, then try again:\n"; \

